File: donors/orchestration-java/ComponentInterface.java
package com.aiassistant.core.orchestration;

import java.util.List;
import java.util.Map;

public interface ComponentInterface {
    
    String getComponentId();
    
    String getComponentName();
    
    List<String> getCapabilities();
    
    void initialize();
    
    void start();
    
    void stop();
    
    ComponentStateSnapshot captureState();
    
    void restoreState(ComponentStateSnapshot snapshot);
    
    Map<String, Object> execute(Map<String, Object> input);
    
    boolean isHealthy();
    
    void heartbeat();
    
    String getStatus();
}

File: android/app/src/main/kotlin/com/ariaagent/mobile/core/orchestration/ComponentInterface.kt
// Ported from donors/orchestration-java/ComponentInterface.java
// Original repo: TITANICBHAI/AI-ASSISTANT-INCOMPLETE
// Changes: idiomatic Kotlin contract, default heartbeat() implementation, suspend lifecycle hooks.

package com.ariaagent.mobile.core.orchestration

/**
 * Contract every component plugged into the [CentralOrchestrator] must satisfy.
 *
 * Implementations should be coroutine-aware: [initialize], [start], and [stop]
 * are suspending so the orchestrator can wire heavy work (model loading, JNI
 * init) onto Dispatchers.IO without blocking the lifecycle.
 *
 * [execute] runs on the caller's dispatcher — it should not block.
 */
interface ComponentInterface {

    val componentId: String

    val componentName: String

    val capabilities: List<String>

    suspend fun initialize()

    suspend fun start()

    suspend fun stop()

    fun captureState(): ComponentStateSnapshot

    fun restoreState(snapshot: ComponentStateSnapshot)

    suspend fun execute(input: Map<String, Any?>): Map<String, Any?>

    fun isHealthy(): Boolean

    fun heartbeat() {
        // Default no-op; HealthMonitor calls recordHeartbeat() on its own.
    }

    fun status(): String
}
